### ISSUE TITLE
Add XM format option

### ISSUE THREAD

**Issue by sapphire-bt** [bunnytrack/umx-converter#3]
As suggested in #1, using XM format may lead to better compression due to the format's use of delta-encoding for samples.

**Comment by sapphire-bt** [comment]
**TL;DR:** it's probably not worth implementing a `.xm` file option. See the tables below for some file size comparisons I've done.

### File Sizes

<table>
	<thead>
		<tr>
			<th>MP3 File</th>
			<th>Duration</th>
			<th>Size (.wav)*</th>
			<th>Size (.it)</th>
			<th>Size (.xm)</th>
			<th>Size (.it.uz)</th>
			<th>Size (.xm.uz)</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>Trance music</td>
			<td>06:27</td>
			<td>16,764 KB</td>
			<td>16,667 KB</td>
			<td>16,668 KB</td>
			<td>8,331 KB</td>
			<td>7,962 KB</td>
		</tr>
		<tr>
			<td>Pop music</td>
			<td>03:36</td>
			<td>9,646 KB</td>
			<td>9,305 KB</td>
			<td>9,306 KB</td>
			<td>4,845 KB</td>
			<td>4,739 KB</td>
		</tr>
		<tr>
			<td>Rock music</td>
			<td>03:56</td>
			<td>10,522 KB</td>
			<td>10,167 KB</td>
			<td>10,167 KB</td>
			<td>7,829 KB</td>
			<td>7,650 KB</td>
		</tr>
		<tr>
			<td>Classical music</td>
			<td>02:10</td>
			<td>2,837 KB</td>
			<td>2,811 KB</td>
			<td>2,811 KB</td>
			<td>653 KB</td>
			<td>651 KB</td>
		</tr>
	</tbody>
</table>

\*8-bit/22 kHz.

### Compression Difference

<table>
	<thead>
		<tr>
			<th>MP3 File</th>
			<th>Size (.it.uz)</th>
			<th>Size (.xm.uz)</th>
			<th colspan="2">Difference (KB)</th>
		</tr>
	</thead>
	<tbody>
		<tr>
			<td>Trance music</td>
			<td>8,331 KB</td>
			<td>7,962 KB</td>
			<td>-369 KB</td>
			<td>-4.43%</td>
		</tr>
		<tr>
			<td>Pop music</td>
			<td>4,845 KB</td>
			<td>4,739 KB</td>
			<td>-106 KB</td>
			<td>-2.19%</td>
		</tr>
		<tr>
			<td>Rock music</td>
			<td>7,829 KB</td>
			<td>7,650 KB</td>
			<td>-179 KB</td>
			<td>-2.28%</td>
		</tr>
		<tr>
			<td>Classical music</td>
			<td>653 KB</td>
			<td>651 KB</td>
			<td>-2 KB</td>
			<td>-0.32%</td>
		</tr>
	</tbody>
</table>

---

Clearly `.xm` does compress slightly better than `.it`, but in my opinion it's not worth the effort of implementing another format just to save a few kilobytes, so I am closing this issue for now.

If anyone feels like adding `.xm` then pull requests are welcome.

Thanks for the suggestion @peterekepeter.

**Comment by peterekepeter** [comment]
Wow, thanks for investigating this! Very cool! Some bytes do get shaved off, but yeah the amount of effort might not be worth it.
